experiments/customui/parent.js

```diff
@@ -250,21 +250,46 @@ var ex_customui = class extends ExtensionCommon.ExtensionAPI {
       return result;
     };
 
+    // Enables dynamic width and fixed heigth for a WebExtension frame
+    const setWebextFrameSizesForSidebar = function(frame, options) {
+      setWebextFrameSizesForBox(frame, options, true, "244px", false, "100%")
+    }
+
     // Enables dynamic height and fixed 100% width for a WebExtension frame
     const setWebextFrameSizesForVerticalBox = function(frame, options) {
-      frame.width = "100%";
-      frame.height = (options.height || 100) + "px";
-      frame.style.display = options.hidden ? "none" : "block";
+      setWebextFrameSizesForBox(frame, options, false, "100%", true, "100px");
+    }
+
+    const setWebextFrameDynamicDimension = function(frame, options, dimensionName, defaultValue) {
+      frame[dimensionName] = (options[dimensionName] || defaultValue) + "px";
       frame.addCustomUILocalOptionsListener(lOptions => {
-        if (typeof lOptions.height === "number") {
-          frame.height = lOptions.height + "px";
-          frame.style.height = frame.height;
+        if (typeof lOptions[dimensionName] === "number") {
+          frame[dimensionName] = lOptions[dimensionName] + "px";
+          frame.style[dimensionName] = frame[dimensionName];
         }
         if (typeof lOptions.hidden === "boolean") {
           frame.style.display = lOptions.hidden ? "none" : "block";
         }
       });
     };
+
+    const setWebextFrameFixedDimension = function(frame, dimensionName, value) {
+      frame[dimensionName] = value;
```

Comment:
You did not also set `frame.style[dimensionName]`. I can't say that I recall why that was needed in the dynamic case, but did you check it is really not needed here?

Reply:
Copied from https://github.com/rsjtdrjgfuzkfg/thunderbird-experiments/blob/11d356aace03b0c5d3a178e2cf98ac4166c073d7/experiments/customui/parent.js#L256 ;)
It works for me...